### ISSUE TITLE
Expose payment flag on public listing page

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-frontend#2080]
Following https://github.com/mozilla/addons-server/issues/4954, we need to expose the payment requirement flag on the public listing page.

Our official policy would be like:
A payment is required to use this add-on or the service this add-on relies on, either upfront, after a trial period or in conjunction with a paid software.

This sounds a bit too wordy to me. I think we could simply and go by:
A payment is required to use this add-on or the service it relies on.

//cc @jvillalobos
//cc @pwalm for UX (on the new desktop pages)
Should we expose it on mobile as well?

**Comment by muffinresearch** [comment]
@wagnerand so to clarify this is really about the need to disclose that a payment is required at some point - it doesn't have anything to do with taking payments?

**Comment by wagnerand** [comment]
Correct. This is *not* about Paypal contributions to the add-on/developer.

**Comment by pwalm** [comment]
I'm designing some badges that this kind of info would fit into nicely.

**Comment by jvillalobos** [comment]
We should support this along with the new review process in Q2 (though I wouldn't block on this). We'll want a checkbox in the New Add-on submission flow, in the Developer Hub, and some indication of this in the listing page. More details pending.

**Comment by wagnerand** [comment]
@muffinresearch we are getting closer on publishing the new review process. As a minimum, could we add those checkboxes to the submission flow and developer hub, so we can ask developers to tick it if necessary?

**Comment by wagnerand** [comment]
Actually, this task is more appropriate for https://github.com/mozilla/addons-server/issues/4954

**Comment by wagnerand** [comment]
No sure if @pwalm already designed UX for this, setting the flag to loop him in...

**Comment by muffinresearch** [comment]
@pwalm did you have any UX for this?

**Comment by pwalm** [comment]
I do not have UX for this yet, but I will explore it this week.

**Comment by pwalm** [comment]
Could be as simple as this? Top info block of the extension detail page:
![screen shot 2018-02-02 at 2 57 25 pm](https://user-images.githubusercontent.com/6797635/35758535-6c68dfe4-0829-11e8-8fad-0890ed09201c.png)

**Comment by wagnerand** [comment]
This looks good to me.

@jvillalobos for product approval.

**Comment by jvillalobos** [comment]
I like the icon and positioning. The copy might need softening, since in most cases there isn't a payment required upfront. Maybe something like "May require payment" or "Includes non-free features". @devaneymoz, any other ideas?

**Comment by devaneymoz** [comment]
What will prompt this badge/message to display on the listing page? Is it developer self-selected, or auto-generated based on identifying payment transaction code? 

As jorge points out, the most common cases are "freemium" models where only portions of the feature set sit behind a pay wall. Given this, perhaps we could clarify further with something like:

"Some features may requires payment"

**Comment by heubergen** [comment]
<img width="725" alt="screen shot 2018-02-15 at 2 00 11 pm" src="https://user-images.githubusercontent.com/5529410/36278204-b31491ca-1258-11e8-8bbc-7c1044ecef7f.png">
There's already an option in the frontend for developer, I think this can/should be used to display the flag.

**Comment by jvillalobos** [comment]
Yeah, it's self-selected by the developer, and reviewers can require developers to set it when it should.

"Some features may require payment" sounds good to me, though it may be a bit too wordy for that section of the listing page (more so in other languages).

**Comment by kumar303** [comment]
Bumping priority because there is an upcoming policy change related to this.

**Comment by bobsilverberg** [comment]
Issue #4520 has been opened to address reformatting the card to reflect the Sketch specs, but as this is a higher priority issue than that, I will tackle adding this badge first, without the change to formatting.

**Comment by vcarciu** [comment]
This is not correctly aligned (to left ) on mobile view : 
![payment](https://user-images.githubusercontent.com/1583842/37213139-d09c7a4c-23b9-11e8-8814-0e931807e3d1.png)

**Comment by bobsilverberg** [comment]
It looks like all of the badges are right-aligned in the small view. See attached:

<img width="374" alt="screenshot 2018-03-09 10 11 31" src="https://user-images.githubusercontent.com/142755/37214228-60b690b4-2382-11e8-9ea0-4c8e0bbb8dfa.png">

It's also how the mock looks in Sketch that is referenced in issue #4520.

So maybe this is intentional, and it's not meant to be left-aligned. @pwalm?

**Comment by kumar303** [comment]
@vcarciu please file a new issue. A good rule of thumb for when to re-open an issue is if the patch has caused such a bad regression that we need to revert the patch. For all other cases, file a new issue.

**Comment by kumar303** [comment]
Also, I think this layout might change a bit when we match the mocks: https://github.com/mozilla/addons-frontend/issues/4520

**Comment by vcarciu** [comment]
Logged another issue, will mark this as verified.
Thanks
